velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -149,6 +151,111 @@ TpchPlan TpchQueryBuilder::getQ1Plan() const {
+      "o_orderdate", "o_shippriority", "o_custkey", "o_orderkey"};
+  std::vector<std::string> customerColumns = {"c_custkey", "c_mktsegment"};
+
+  auto lineitemSelectedRowType = getRowType(kLineitem, lineitemColumns);
+  const auto& lineitemFileColumns = getFileColumnNames(kLineitem);
+  auto ordersSelectedRowType = getRowType(kOrders, ordersColumns);
+  const auto& ordersFileColumns = getFileColumnNames(kOrders);
+  auto customerSelectedRowType = getRowType(kCustomer, customerColumns);
+  const auto& customerFileColumns = getFileColumnNames(kCustomer);
+
+  const auto orderDate = "o_orderdate";
+  const auto shipDate = "l_shipdate";
+  std::string orderDateFilter;
+  std::string shipDateFilter;
+  std::string customerFilter;
+
+  // DWRF does not support Date type. Use Varchar instead.
+  if (ordersSelectedRowType->findChild(orderDate)->isVarchar()) {
+    orderDateFilter = "o_orderdate < '1995-03-15'";
+  } else {
+    orderDateFilter = "o_orderdate < '1995-03-15'::DATE";
+  }
+
+  if (lineitemSelectedRowType->findChild(shipDate)->isVarchar()) {
+    shipDateFilter = "l_shipdate > '1995-03-15'";
+  } else {
+    shipDateFilter = "l_shipdate > '1995-03-15'::DATE";
+  }
+
+  customerFilter = "c_mktsegment = 'BUILDING'";
+
+  auto planNodeIdGenerator = std::make_shared<PlanNodeIdGenerator>();
+  core::PlanNodeId lineitemPlanNodeId;
+  core::PlanNodeId ordersPlanNodeId;
+  core::PlanNodeId customerPlanNodeId;
+
+  auto customers = PlanBuilder(planNodeIdGenerator)
+                       .tableScan(
+                           kCustomer,
+                           customerSelectedRowType,
+                           customerFileColumns,
+                           {customerFilter})
+                       .capturePlanNodeId(customerPlanNodeId)
+                       .planNode();
+
+  auto custkeyJoinNode =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(
+              kOrders,
+              ordersSelectedRowType,
+              ordersFileColumns,
+              {orderDateFilter})
+          .capturePlanNodeId(ordersPlanNodeId)
+          .hashJoin(
+              {"o_custkey"},
+              {"c_custkey"},
+              customers,
+              "",
+              {"o_orderdate", "o_shippriority", "o_orderkey"})
+          .planNode();
+
+  auto plan =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(
+              kLineitem,
+              lineitemSelectedRowType,
+              lineitemFileColumns,
+              {shipDateFilter})
+          .capturePlanNodeId(lineitemPlanNodeId)
+          .project(
+              {"l_extendedprice * (1.0 - l_discount) AS part_revenue",
+               "l_orderkey"})
+          .hashJoin(
+              {"l_orderkey"},
+              {"o_orderkey"},
+              custkeyJoinNode,
+              "",
+              {"l_orderkey", "o_orderdate", "o_shippriority", "part_revenue"})
+          .localPartition({})
+          .finalAggregation(
```

Comment:
It is unusual to have final aggregation without partial aggregation. One option is to add partial aggregation before local exchange or change this to singleAggregation.

Reply:
Thanks for the feedback, I have added partialAggregation before the finalAggregation is done.